### PR TITLE
ZCL_ABAPGIT_HASH: Change FM call to method call

### LINKED ISSUE #3977: ZCL_ABAPGIT_HASH, replace call of CALCULATE_HASH_FOR_RAW
replace call of function module `CALCULATE_HASH_FOR_RAW` with call to `CL_ABAP_HMAC`, it should exist in 702

```abap
DATA hash TYPE string.
DATA xstr TYPE xstring.
DATA input TYPE xstring VALUE 'AA11'.

cl_abap_hmac=>calculate_hmac_for_raw(
  EXPORTING
    if_key        = xstr
    if_data       = input
  IMPORTING
    ef_hmacstring = hash ).

WRITE / to_lower( hash ).

DATA(res) = zcl_abapgit_hash=>sha1_raw( input ).
WRITE / res.
```

### PR DESCRIPTION
**Pull request by g-back**
closes #3977 

- change FM call to method call
- add unit test for `sha1_raw`

### CODE REVIEW

File: src/utils/zcl_abapgit_hash.clas.testclasses.abap [hunk 1]
```diff
@@ -39,4 +40,18 @@ CLASS ltcl_test IMPLEMENTATION.
 
   ENDMETHOD.
 
+  METHOD sha1_raw.
+
+    DATA: lv_sha1  TYPE zif_abapgit_definitions=>ty_sha1,
+          lv_input TYPE xstring.
+
+    lv_input = 'LOREM_IPSUM'.
```
Comment (larshp) [inline src/utils/zcl_abapgit_hash.clas.testclasses.abap:48]: suggest also adding a valid test case, the XSTRING data only allows up to "F", hence lv_input becomes empty

**Review by larshp — CHANGES_REQUESTED** [review]

**Comment by larshp** [comment]
thanks 👍

**Review by larshp — APPROVED** [review]
